content/rust/github/publish-github-release-multi-arch.md

```diff
@@ -0,0 +1,419 @@
+
+{{< highlight text "linenos=table,linenostart=9,hl_lines=5-11" >}}
+steps:
+- name: build
+  image: rust-multiarch
+  commands:
+    - mkdir assets
+    - cargo build
+    - cp target/debug/hello-world assets/hello-world-amd64
+    - cargo build --target=aarch64-unknown-linux-gnu
+    - cp target/aarch64-unknown-linux-gnu/debug/hello-world assets/hello-world-aarch64
+    - cargo build --target=armv7-unknown-linux-gnueabihf
+    - cp target/armv7-unknown-linux-gnueabihf/debug/hello-world assets/hello-world-armv7
+{{< / highlight >}}
+
+## Execution
+
+The Pipeline will execute every time you push code to your repository.
+You can login to Drone and inspect the Pipeline results:
+
+![Build Pipeline](/screenshots/drone-build.png)
+
+# Continuous Delivery
+
+The final step is to automatically build the project and create a GitHub release when you have created a git tag.
+In order to use the GitHub API you need to provide your credentials.
+
+## Secrets
+
+For security reasons you do not want to store your credentials in the pipeline configuration file.
+Instead you can securely store your secrets in the Drone database, adding your GitHub token via the repository settings screen.
+
+![Manage Secret](/screenshots/github-token-secret.jpg)
+
+## Settings
+
+Drone has a robust plugin system, including a plugin to create GitHub releases.
+Add a new step to the pipeline and configure [the GitHub plugin](http://plugins.drone.io/drone-plugins/drone-github-release/).
+
+{{< highlight text "linenos=table,linenostart=12,hl_lines=99" >}}
+- name: publish
+  image: plugins/github-release
+  settings:
+    api_key: 
+      from_secret: github_token
+    files: assets/*
+    title: Release Title
+  when:
+    event: tag
+{{< / highlight >}}
+
+### The `image` Attribute
+
+Defines the image used to publish the GitHub release.
+
+{{< alert info >}}
+Drone plugins are Docker images that perform pre-defined tasks.
+Choose from hundreds of community plugins or create your own.
+{{< / alert >}}
+
+{{< highlight text "linenos=table,linenostart=12,hl_lines=2" >}}
+- name: publish
+  image: plugins/github-release
+  settings:
+    api_key: 
+      from_secret: github_token
+    files: assets/*
+    title: Release Title
+  when:
+    event: tag
+{{< / highlight >}}
+
+### The `api_key` Attribute
+
+Provides the GitHub API token used to authenticate, sourced from the named secret.
+
+{{< highlight text "linenos=table,linenostart=12,hl_lines=4-5" >}}
+- name: publish
+  image: plugins/github-release
+  settings:
+    api_key: 
+      from_secret: github_token
+    files: assets/*
+    title: Release Title
+  when:
+    event: tag
+{{< / highlight >}}
+
+### The `files` Attribute
+
+This attribute defines the path to the binaries you wish to upload.
+
+{{< highlight text "linenos=table,linenostart=12,hl_lines=6" >}}
+- name: publish
+  image: plugins/github-release
+  settings:
+    api_key: 
+      from_secret: github_token
+    files: assets/*
```

Comment:
This attribute could also be a list of paths, which supports simple globbing, e.g. `assets/**/foo-*`.